src/bp/ui-studio/src/web/components/Shared/ElementsList/index.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { ListGroupItem, Glyphicon } from 'react-bootstrap'
+import { InputElement } from './InputElement'
+import style from './style.scss'
+
+export default class ElementsList extends React.Component {
+  elementInputRef = React.createRef()
+  state = {
+    inputValue: '',
+    error: undefined,
+    editElementIndex: undefined
+  }
+
+  toggleEditMode = index => {
+    this.setState({ editElementIndex: index })
+  }
+
+  handleEditEnter = (element, index) => {
+    this.setState({ editElementIndex: undefined })
+    this.props.update(element, index)
+  }
+
+  renderElement = (element, index) => {
+    if (this.props.renderElement) {
+      return this.props.renderElement(element)
+    }
```

Comment:
not sure what this does, is it for a custom element renderer?

Reply:
yeah this allows to override `renderElement`

Reply:
id suggest a more meaningful name, like this.props.customElementRenderer